File: database/fix-missing-prices.py
#!/usr/bin/env python3
"""
Script de correction du menu.json - Ajoute priceSolo=0 aux produits sans prix
"""

import json
import os
from datetime import datetime

menu_path = '/home/user/snackapp-/config/menu.json'

# Backup avant modification
backup_path = f"{menu_path}.backup-{datetime.now().strftime('%Y%m%d-%H%M%S')}"
os.system(f"cp {menu_path} {backup_path}")
print(f"✅ Backup créé: {backup_path}")

# Charger menu.json
with open(menu_path, 'r', encoding='utf-8') as f:
    menu_data = json.load(f)

# Corriger les produits sans prix
fixed_count = 0
for category in menu_data.get('menu', {}).get('categories', []):
    for item in category.get('items', []):
        # Si le produit n'a pas de priceSolo, lui donner 0
        if 'priceSolo' not in item or item['priceSolo'] is None:
            item['priceSolo'] = 0
            fixed_count += 1
            print(f"✅ Corrigé: {item.get('name', 'sans nom')} → priceSolo = 0")

# Sauvegarder
with open(menu_path, 'w', encoding='utf-8') as f:
    json.dump(menu_data, f, ensure_ascii=False, indent=2)

print(f"\n✅ {fixed_count} produits corrigés")
print(f"✅ menu.json sauvegardé")

File: snackup/backend/fix-missing-prices.php
<?php
/**
 * Script de correction menu.json - Ajoute priceSolo=0 aux produits sans prix
 * Résout le problème: API 400 Bad Request + Admin panel vide
 */

$menuPath = __DIR__ . '/../config/menu.json';

// Backup avant modification
$backupPath = $menuPath . '.backup-' . date('Ymd-His');
copy($menuPath, $backupPath);
echo "✅ Backup créé: $backupPath\n\n";

// Charger menu.json
$menuData = json_decode(file_get_contents($menuPath), true);

if (!$menuData) {
    die("❌ Erreur: Impossible de lire menu.json\n");
}

// Corriger les produits sans prix
$fixedCount = 0;
foreach ($menuData['menu']['categories'] as &$category) {
    foreach ($category['items'] as &$item) {
        // Si le produit n'a pas de priceSolo ou est null, lui donner 0
        if (!isset($item['priceSolo']) || $item['priceSolo'] === null) {
            $item['priceSolo'] = 0;
            $fixedCount++;
            echo "✅ Corrigé: " . ($item['name'] ?? 'sans nom') . " → priceSolo = 0\n";
        }
    }
}

// Sauvegarder
$json = json_encode($menuData, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE | JSON_UNESCAPED_SLASHES);
file_put_contents($menuPath, $json);

echo "\n";
echo "✅ $fixedCount produits corrigés\n";
echo "✅ menu.json sauvegardé\n";
echo "\n";
echo "🔄 Rechargez maintenant l'admin panel dans votre navigateur\n";
